plugins/linear-genome-view/src/BaseLinearDisplay/models/BaseLinearDisplayModel.tsx

```diff
@@ -245,11 +298,100 @@ export const BaseLinearDisplay = types
       self.contextMenuFeature = feature
     },
   }))
+  .actions(self => {
+    const { reload: superReload } = self
+
+    return {
+      async reload() {
+        self.setError()
+        const aborter = new AbortController()
+        const view = getContainingView(self) as LGV
+        if (!view.initialized) {
+          return
+        }
+
+        let stats
+        if (view.staticBlocks.contentBlocks[0]) {
+          try {
+            stats = await self.getGlobalStats(
+              view.staticBlocks.contentBlocks[0],
+              { signal: aborter.signal },
+            )
+
+            if (isAlive(self)) {
+              self.updateGlobalStats(stats)
+              superReload()
+            } else {
+              return
+            }
+          } catch (e) {
+            self.setError(e)
+            self.setStatsStatus('error')
+          }
+        }
+      },
+      afterAttach() {
+        addDisposer(
+          self,
+          autorun(
+            async () => {
+              try {
+                const aborter = new AbortController()
+                const view = getContainingView(self) as LGV
+
+                // whereever it's skipping stats, need to set status back to none
+                if (!view.initialized) {
+                  return
+                }
+
+                if (
+                  view &&
+                  self.globalStats &&
+                  self.globalStats.featureDensity > self.maxFeatureScreenDensity
+                ) {
+                  return
+                }
+
+                if (view.staticBlocks.contentBlocks[0]) {
+                  const stats = await self.getGlobalStats(
+                    view.staticBlocks.contentBlocks[0],
+                    {
+                      signal: aborter.signal,
+                    },
+                  )
+
+                  if (isAlive(self)) {
+                    self.updateGlobalStats(stats)
+                  } else {
+                    return
+                  }
+                }
+              } catch (e) {
+                if (!isAbortException(e) && isAlive(self)) {
+                  console.error(e)
+                  self.setError(e)
+                  self.setStatsStatus('error')
+                }
+              }
+            },
+            { delay: 1000 },
+          ),
+        )
+      },
+    }
+  })
   .views(self => ({
     regionCannotBeRenderedText(_region: Region) {
       const view = getContainingView(self) as LinearGenomeViewModel
-      if (view && view.bpPerPx > self.maxViewBpPerPx) {
-        return 'Zoom in to see features'
+      if (self.statsStatus === 'error') {
+        return 'Force load to see features'
```

Comment:
it seems if there was an error, we should report the raw error string to the user maybe?